virtme_ng/run.py

```diff
@@ -49,20 +50,38 @@ def check_call_cmd(command, quiet=False, dry_run=False):
         stdout_fd = process.stdout.fileno()
         stderr_fd = process.stderr.fileno()
 
+        stdout_open = True
+        stderr_open = True
+
         # Use select to poll for new data in the file descriptors
-        while process.poll() is None:
```

Comment:
just to be sure I understand the issue: was the problem before that this could return `None` while there were still data to read?

Do you need the rest of the modifications to avoid calling `select()` with a `fd` where we had a `EOF` even if it has not been closed?

(on the other hand, the new code is not very complex, I'm happy to have this merged :) )

Reply:
Right, the problem is that `process.poll()` can return `None` if either stdout or stderr are closed, but we need to consume all data from both.

And yes, we can probably compact the code a bit more and always pass both fds to select(), even if one is EOF'd the behavior should be the same. Let me try to make the code a bit more compact.

Reply:
thank you for the explanation and the update!